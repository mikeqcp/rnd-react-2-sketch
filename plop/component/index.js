const path = require('path');

const templatesPath = path.join(__dirname, 'templates');

module.exports = (plop) => {
  const containerDirectory = 'src/{{ directory }}/{{ camelCase name }}';
  plop.setGenerator('simple component', {
    description: 'Simple component',
    prompts: [{
      type: 'input',
      name: 'name',
      message: 'Name:',
    }, {
      type: 'directory',
      name: 'directory',
      basePath: 'src',
      message: 'Component\'s directory:',
    }],
    actions: [{
      type: 'add',
      path: `${containerDirectory}/{{ camelCase name }}.component.js`,
      templateFile: path.join(templatesPath, 'component.hbs'),
    }, {
      type: 'add',
      path: `${containerDirectory}/{{ camelCase name }}.styles.js`,
      templateFile: path.join(templatesPath, 'styles.hbs'),
    }, {
      type: 'add',
      path: `${containerDirectory}/index.js`,
      templateFile: path.join(templatesPath, 'index.hbs'),
    }],
  });
};
