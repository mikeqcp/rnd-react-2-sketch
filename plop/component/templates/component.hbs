import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Container } from './{{ camelCase name }}.styles';

export class {{ pascalCase name }} extends PureComponent {
  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

{{ pascalCase name }}.propTypes = {
};
