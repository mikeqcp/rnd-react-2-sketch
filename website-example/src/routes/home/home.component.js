import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

import messages from './home.messages';
import { Container } from './home.styles';


export class Home extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
  };

  render() {
    return (
      <Container>
        <Helmet
          title={this.props.intl.formatMessage(messages.pageTitle)}
        />
      </Container>
    );
  }
}
