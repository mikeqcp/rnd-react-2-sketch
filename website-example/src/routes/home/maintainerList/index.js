export { MaintainerList } from './maintainerList.component';
