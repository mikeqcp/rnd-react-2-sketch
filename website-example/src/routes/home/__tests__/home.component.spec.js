import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { fromJS } from 'immutable';

import { Home } from '../home.component';
import { DEFAULT_LOCALE, LOCALES } from '../../../i18n';


describe('Home: Component', () => {
  const defaultProps = {
    items: fromJS([{ id: 1 }, { id: 2 }, { id: 3 }]),
    intl: { formatMessage: ({ id }) => id },
    language: DEFAULT_LOCALE,
    fetchMaintainers: () => {},
  };

  const component = (props) => (
    <Home {...defaultProps} {...props} />
  );

  it('should render correctly', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should fetch maintainers on mount', () => {
    const fetchMaintainers = spy();
    shallow(component({ fetchMaintainers }));

    expect(fetchMaintainers).to.have.been.calledOnce;
    expect(fetchMaintainers).to.have.been.calledWith(DEFAULT_LOCALE);
  });

  it('should fetch maintainers when language changes', () => {
    const fetchMaintainers = spy();
    const wrapper = shallow(component({ fetchMaintainers }));

    fetchMaintainers.resetHistory();
    wrapper.setProps({ language: LOCALES.POLISH });

    expect(fetchMaintainers).to.have.been.calledOnce;
    expect(fetchMaintainers).to.have.been.calledWith(LOCALES.POLISH);
  });
});
