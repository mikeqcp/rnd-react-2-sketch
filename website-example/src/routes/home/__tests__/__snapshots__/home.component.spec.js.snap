// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Component should render correctly 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="home.pageTitle"
  />
  <styled.h1>
    <FormattedMessage
      defaultMessage="Hello World!"
      id="home.welcome"
      values={Object {}}
    />
  </styled.h1>
  <Styled(ReactComponent) />
  <InjectIntl(MaintainerListComponent)
    items={
      Immutable.List [
        Immutable.Map {
          "id": 1,
        },
        Immutable.Map {
          "id": 2,
        },
        Immutable.Map {
          "id": 3,
        },
      ]
    }
  />
  <Connect(InjectIntl(withRouter(LanguageSwitcher))) />
</styled.div>
`;
